### ISSUE TITLE
Archived changed to skip

### ISSUE THREAD

**Issue by randomevents** [evilhero/mylar#1732]
Since updating in the past couple of days (currently on  8d199d3ec7e45bf8d2845822ee6126fef1ad33a0 (development)), the Db Updater processing running through the entries (every 5 minutes?) and marking all 'Archived' comics to 'Skipped' (In the process of converting the data to CV, I changed the status of 40 issues.).

For context, all my downloaded issues are marked as Archived and move to different filing system. Every so often Mylar randomly picks a series and tries to redownload everything?

Changing them back to Archive just ends up as Skipped next time the updater runs.

[(u'DB Updater', u'2017-09-23 17:15:31.601000', 1506186931.601, u'2017-09-23 17:20:31.602000', 1506187231.602, u'True', None, None, u'Waiting')

**Comment by barbequesauce** [comment]
Glad you posted this. I’ve been trying to understand what was happening too. It’s not just archived issues - I have the same problem with downloaded.

I’ve noticed the following line in the log before it starts getting weird. I haven’t dug into the code yet; it’s only begun in the last week or so...

23-Sep-2017 16:27:41 - WARNING :: Thread-12 : Something is out of whack somewhere with the series

**Comment by evilhero** [comment]
Thr out of whack happens when there are more 'have' issues than total issues (as per the main page) - these are colour coded like a yellow/orange colour to indicate a problem.

There was an issue this week where everything a series got refreshed it would either assume there was annuals or not depending on the previous state (only for annual integration). Also this week when a series got refreshed it marked all issues as None regardless of the status which basically screwed up the counts amongst other things.

If you guys are both up to date on dev, and you've refreshed a series since you've been up to date and still experience the problem, then it's a new issue that's caused by the old ones fix/bug.

I've been afk most of the day today, but will try and look at this in the next few hrs to see at least what's happening...

**Comment by barbequesauce** [comment]
I’m current (ad33a0). Last occurrence was on Teen Titans. 12 files in directory, no annuals. Display currently shows episode 1 as skipped, and 2-12 as downloaded. “Refresh comic” didn’t seem to do anything.

**Comment by evilhero** [comment]
So is that the correct count tho (11/12)? Are the issues that say Downloaded actually present in the directory still?

**Comment by barbequesauce** [comment]
All 12 are there - 11 downloaded and 1 skipped.

**Comment by evilhero** [comment]
Weird, can you post the debug log when you do a Recheck Files? I'm wondering if it's something with the file parser, or if it's only occuring when you do a Refresh Files... (not sure if this is identical to the Archive problem, but both need to get addressed regardless)

**Comment by randomevents** [comment]
Moving this to its own issue "For context, all my downloaded issues are marked as Archived and move to different filing system. Every so often Mylar randomly picks a series and tries to redownload everything?"

Okay, I did a file scan and it doesn't mess with the status, however the CV scan does. It's marking them as out of whack because the CV count is always one less than Mylar's count. So there's an adjustment missing?

24-Sep-2017 02:09:15 - DEBUG   :: CP Server Thread-6 : Refreshing comic: [u'17993']
24-Sep-2017 02:09:15 - INFO    :: Thread-37 : Starting update for 1 active comics
24-Sep-2017 02:09:15 - DEBUG   :: Thread-37 : Refreshing: Invincible (2003) [17993]
24-Sep-2017 02:09:15 - DEBUG   :: Thread-37 : CV_OneTimer option enabled...
24-Sep-2017 02:09:15 - DEBUG   :: Thread-37 : Gathering the status of all issues for the series.
24-Sep-2017 02:09:15 - DEBUG   :: Thread-37 : Checking how out of whack the series is.
24-Sep-2017 02:09:15 - DEBUG   :: Thread-37 : checking annual db
24-Sep-2017 02:09:15 - DEBUG   :: Thread-37 : Refreshing the series and pulling in new data using only CV.
24-Sep-2017 02:09:15 - INFO    :: Thread-37 : [DIRECTORY-CHECK] Found comic directory: \\fragments\shares\Disk - Media\Media - Comics\Comics - Process\Invincible (2003)
24-Sep-2017 02:09:16 - DEBUG   :: Thread-14 : there are 0 search results...
24-Sep-2017 02:09:16 - DEBUG   :: Thread-14 : [IMPORTER-ANNUAL] - No results, removing the year from the agenda and re-querying.
24-Sep-2017 02:09:16 - DEBUG   :: Thread-14 : Default search mode - this will match on ALL search words
24-Sep-2017 02:09:17 - DEBUG   :: Thread-37 : number of issues I counted: 141
24-Sep-2017 02:09:17 - DEBUG   :: Thread-37 : number of issues CV says it has: 140
24-Sep-2017 02:09:17 - DEBUG   :: Thread-37 : vari is set to: yes
24-Sep-2017 02:09:18 - DEBUG   :: Thread-37 : comic[ComicVersion]:None
24-Sep-2017 02:09:18 - INFO    :: Thread-37 : Now adding/updating: Invincible
24-Sep-2017 02:09:18 - INFO    :: Thread-37 : 2003
24-Sep-2017 02:09:18 - INFO    :: Thread-37 : Sucessfully retrieved details for Invincible
24-Sep-2017 02:09:18 - INFO    :: Thread-37 : Previous version detected as None - seeing if update required
24-Sep-2017 02:09:18 - INFO    :: Thread-37 : Directory (\\fragments\shares\Disk - Media\Media - Comics\Comics - Process\Invincible (2003)) already exists! Continuing...
24-Sep-2017 02:09:19 - DEBUG   :: Thread-14 : there are 0 search results...

**Comment by randomevents** [comment]
updater.py line 242:
if issuenew['Status'] is None:

should be some variation of -:

if issuenew['Status'] == None and (issue['Status'] != 'Downloaded' and issue['Status'] != 'Archived' and issue['Status'] != 'Ignored' and issue['Status'] != 'Skipped'):

**Comment by barbequesauce** [comment]
yeah, looking a little closer it seems to be about the name parser. Teen Titans has 12 files in the directory; GUI displays 1 skipped and 12 downloaded. Here's the file names:

```
Teen Titans 002 (2016) (Digital) (Thornn-Empire).cbr
Teen Titans 003 (2016) (Digital) (Thornn-Empire).cbr
Teen Titans 004 (2017) (Digital) (Thornn-Empire).cbr
Teen Titans 005 (2017) (Digital) (Thornn-Empire).cbr
Teen Titans 006 (2017) (Digital) (Thornn-Empire).cbr
Teen Titans 007 (2017) (2 covers) (digital) (Minutemen-Thoth).cbz
Teen Titans 008 (2017) (2 covers) (digital) (Minutemen-Thoth).cbz
Teen Titans 009 (2017) (2 covers) (digital) (Minutemen-Thoth).cbz
Teen Titans 010 (2017) (2 covers) (digital) (Minutemen-Thoth).cbz
Teen Titans 011 (2017) (2 covers) (digital) (Minutemen-Thoth).cbz
Teen Titans 012 (2017) (2 covers) (digital) (Minutemen-Thoth).cbz
Teen Titans - Rebirth 001 (2016) (digital) (Oroboros-DCP).cbr
```


Similarly, GUI says 6 downloaded/7 skipped for Doctor Aphra; here's the directory:

```
Doctor Aphra 001 (2017) (6 covers) (digital) (Minutemen-Midas).cbr
Doctor Aphra 002 (2017) (4 covers) (digital) (Minutemen-Midas).cbr
Doctor Aphra 003 (2017) (3 covers) (digital) (Minutemen-Midas).cbr
Doctor Aphra 004 (2017) (2 covers) (digital) (Minutemen-Midas).cbr
Doctor Aphra 005 (2017) (Digital) (Kileko-Empire).cbr
Doctor Aphra 006 (2017) (3 covers) (digital) (Minutemen-Midas).cbr
Doctor Aphra 007 (2017) (digital) (Minutemen-Midas).cbr
Star Wars - Doctor Aphra 008 (2017) (digital) (Minutemen-Midas).cbr
Star Wars - Doctor Aphra 009 (2017) (digital) (Minutemen-Midas).cbr
Star Wars - Doctor Aphra 010 (2017) (Digital) (Kileko-Empire).cbr
Star Wars - Doctor Aphra 011 (2017) (digital) (Minutemen-Midas).cbr
Star Wars - Doctor Aphra 012 (2017) (digital) (Minutemen-Midas).cbr
Star Wars - Doctor Aphra Annual 01 (2017) (digital) (Minutemen-Midas).cbr
```

**Comment by barbequesauce** [comment]
I'm thinking my issue is unrelated to @randomevents...

**Comment by evilhero** [comment]
Yeah looks Ike yours is actually naming conventions.

Teen titans series wouldn't pick up 'Teen Titans - Rebirth' as it's not the same series (plus it doesn't name match). If you want it to pick it up, you'd have to either rename it to 'Teen Titans 001...' or use the Alternate Search Name and put 'Teen Titans - Rebirth', then on the next Recheck Files it would pick it up.

Star Wars is the same - it only will match on what the series I according to CV, anything other than that you'll have to use the Alternate Search Name to get them to search/post-process/refresh properly (ie. Alternate Search Name of 'Star Wars - Doctor Aphra').

I just noticed that some series on CV have some aliases actually listed now so that I can prepopulate the Alternate Search Name with those and hopefully avoid some of these issues...

**Comment by barbequesauce** [comment]
SW Aphra makes sense to me - but Rebirth was the subtitle to all of the “reset” titles’ issue #1’s (hope that makes sense). I wouldn’t expect Mylar to accommodate, but I would expect CV to have an alias accordingly...

**Comment by evilhero** [comment]
@randomevents: yeah the dB update process that runs every 5 minutes has different parameters than a simple refresh on a series due to how it's initiated.

So when the status goes to None, it resets all the info it had on the given issue - meaning archive history as well. The None glitch for fixed, but it wouldn't be able to put things back into an Archived status from None since that's not the default setting (ie. It has no basis to assume that an issue was Archived prior to a scan, especially if the status is showing None/Skipped).

There are also some other things to consider in regards to the workflow, but like I mentioned while it worked for you, it won't work for a vast majority that don't use the Archive feature (trust me, you think you think of all possibilities, but there's always some that pop up that are never considered lol)

I'll take your change under review (thatlbks for that regardless), and I'm going to have to do some more testing with the status' as if it's just the dB updater that's causing the problem it helps to narrow down  things now.

**Comment by evilhero** [comment]
@barbequesauce : yep, I'm aware of the rebirth stuff - but all the titles that were as 'series - rebirth' were actually one-shots. The given series were rebooted like the next month or something, and all had new issue #1's under the new title (with a subtitle of the DC rebirth moniker, but it's not actually referred to in the indica or anything).

Search on CV for both 'teen titans rebirth' and 'teen titans' and you'll see exactly what I'm talking about ;)

**Comment by randomevents** [comment]
@evilhero Looking at the history, it was the switch from (if newVAL['Status'] == None:) to (if issuenew['Status'] is None:) is the start of the issue.

What's interesting is the block that starts at line 221 (if issuenew['Status'] == None and (issue['Status'] == 'Downloaded' or issue['Status'] == 'Archived'):) and ends at 240. My suggestion for 242 is based on the actions of the else for (if mylar.AUTOWANT_ALL:) in that block (newVAL = {"Status":  "Skipped"}), negating the decisions made from for the (221 to 240). 

I took a second to test it with a downloaded files scenario and reverted back to the original code, and it works like it reads. Assuming issuenew['Status'] == None:
-It throws out everything from block 221-240
-Uses AUTOWANT_ALL (Automatically Mark All Issues as Wanted) with timing to set the status to Wanted or Skipped
-Uses the File Scanner to repopulate the Download Status.

In a 7 issue series that originally had two archived, two downloaded and two wanted and one snatched, the Archived, Snatched and Wanted switched to Skipped, because I have AUTOWANT_ALL as false to stop the mass downloads and API hits that have been happening this weekend (I hadn't updated in a month or so before Friday). If I switch it back on a rerun it, those Skippeds become Wanteds.

I'm guessing only those that are organizing their collection with additional steps through Comic Rack and the like would be the only ones that saw this. 

Long story short, it's just an if statement that inadvertently negated the if statement before it.

**Comment by evilhero** [comment]
Yeah, this was all due to one simple change that happened to alter the update workflow completely and now it's trying to piecemeal stuff back around the new workflow in order to make things in line again. The initial fix was to stop issue data being deleted if the refresh/update process failed for some reason (ie. CV API limit hit, directory not accessible, issue errors, etc).

I just tested your code patch, and while it worked for you - not so much for me (yes, I can duplicate it to a degree). It does seem to only apply to the db updater scheduler running - but in my case, I don't use 'Mark All Issues as Wanted', but do have the 'Mark All Upcoming Issues as Wanted' enabled, which is probably what's causing the discrepancy in our results.  

With your code change in place, it does keep Wanted issues as Wanted (in some cases, haven't tracked this down yet, but for some reason a few series were still losing the Wanted status but that may be something entirely else), but the Snatched issues revert back to Skipped. Once they've been set to Skipped, the status of Snatched is lost (I'm pretty sure the linking to post-process these formerly Snatched issues should still be intact, but again haven't tested that). This would then mark All issues that were previously Snatched but not post-processed as either Wanted on a future run (if you had 'Mark All Upcoming' as an example, or even the auto-want all and could potentially lead to more unnecessary downloads.

What it really should be doing is retaining the previous status if it's anything other than Skipped (since that would be the default) on each refresh - but it's not completely doing that atm.

I"ll still push out your patch, as regardless of it not being a complete fix for all uses atm - it works, and it will lower the re-downloads being triggered for some as opposed to everyone and their dog getting affected which give some additional time to work thru the issue that I'm experiencing with it now...

**Comment by randomevents** [comment]
The Snatched issue would be because of an oversight in my quick code -:

if issuenew['Status'] == None and (issue['Status'] != 'Downloaded' and issue['Status'] != 'Archived' and issue['Status'] != 'Ignored' and issue['Status'] != 'Skipped'  and issue['Status'] != 'Snatched'):

I overdid it (and still missed something) because I assumed that I was missing something and that using (if issuenew['Status'] == None and (issue['Status'] == 'Skipped')) was wrong somehow?

Oh wait, it would be. It's have to be (if issuenew['Status'] == None and (issue['Status'] == 'Skipped' or issue['Status'] == 'Wanted')) so it can handle changes to AUTOWANT_ALL. Maybe with an addition "or" for issue['Status'] == 'None' (or something else it shouldn't be).

What's weird is that I do have Upcoming Checked?
I'm guessing that you also need to handle Ignored in the 221 block before the else at 238?

On a side note, I understand your frustration. I'm at a point in a coding project for work, where I've realized that patches and tweaks don't do it anymore, and I need to do an bottom up overhaul because there's so many bits and pieces that rely on the section in question. Looking at this was more a exercise in procrastination then a need to fix something. If you're welcome to help, I'm willing to fork this instead of the tracing I was doing with notepad++ and every source code file open.

**Comment by evilhero** [comment]
Yeah, I just finished testing (just before I saw your new post) with the `if all([issuenew['Status'] == None, issue['Status'] == 'Skipped'])` and it held the status' at that point regardless of previous states, aside from marking Skipped issues as Wanted accordingly due to AUTOWANT_ALL or Mark Upcoming.

I don't think the issue['Status'] would need to include Wanted since it would already be marked as AUTO-WANT from a previous state (meaning issue['Status'] should already be Wanted). Unless I'm missing something (I'm probably doing too many things at once here as usual), as the previous if-statement should catch anything different?

I'm always open to PR's and code-assists - this has always been just a hobby that I started out as way to learn python and because of my love for comics. I honestly never expected to be doing it to the level where it has reached now, let alone the number of users that actually use it and support it. But there are times when my free time isn't as 'free' as I would like to deal with some really important issues with the code, and then the complete base-changes take longer than normal in some cases due to just that same exact reason ;)  Life can be just...life sometimes.
